packages/zend-ldap/library/Zend/Ldap/Collection/Iterator/Default.php

```diff
@@ -339,4 +351,65 @@ protected function _isResultEntry($resource)
         
         return $resource instanceof \LDAP\ResultEntry;
     }
+    
+    /**
+     * Set a sorting-algorithm for this iterator
+     *
+     * The callable has to accept two parameters that will be compared.
+     *
+     * @param callable $sortFunction The algorithm to be used for sorting
+     * @return self Provides a fluent interface
+     */
+    public function setSortFunction(callable $sortFunction)
+    {
+        $this->sortFunction = $sortFunction;
+
+        return $this;
+    }
+
+    /**
+     * Sort the iterator
+     *
+     * Sorting is done using the set sortFunction which is by default strnatcasecmp.
+     *
+     * The attribute is determined by lowercasing everything.
+     *
+     * The sort-value will be the first value of the attribute.
+     *
+     * @param string $sortAttribute The attribute to sort by. If not given the
+     *                              value set via setSortAttribute is used.
+     * @return void
+     */
+    public function sort($sortAttribute)
+    {
+        foreach ($this->entries as $key => $entry) {
+            $attributes = ldap_get_attributes(
+                $this->_ldap->getResource(),
+                $entry['resource']
+            );
+
+            $attributes = array_change_key_case($attributes, CASE_LOWER);
+
+            if (isset($attributes[$sortAttribute][0])) {
+                $this->entries[$key]['sortValue'] =
+                    $attributes[$sortAttribute][0];
+            }
+        }
+
+        $sortFunction = $this->sortFunction;
+        $sorted       = usort(
+            $this->entries,
+            static function($a, $b) use
```

Comment:
remove `static`, requires php 5.4

Reply:
@glensc thanks, I was wondering how to make it PHP 5.3 compatible.

Reply:
@glensc Do you know why this pull fails? I am unable to debug this. Any help would be appreciated.

Reply:
@marcing please debug why `$node` is `null` on https://github.com/zf1s/zf1/blob/ba99db6b24d279debb1b695be9bdb516968e2855/tests/Zend/Ldap/Node/OnlineTest.php#L58 .

Reply:
@falkenhawk I can't reproduce this error on local ldap instance.